mdsuite/calculators/calculator.py

```diff
@@ -364,10 +378,26 @@ def _msd_operation(ensemble: tf.Tensor, square: bool = True):
                 Mean square displacement.
         """
         if square:
-            return tf.math.squared_difference(ensemble, ensemble[:, None, 0])
+            return tf.math.squared_difference(np.array(ensemble)[:, self.tau_values], ensemble[:, None, 0])
```

Comment:
I'd suggest for better performance that we don't use numpy here.
If we need the indexing I'd suggest using `tf.gather` or `tf.gather_nd` for this

Reply:
Yeah this is okay but of the three outcomes only two require the gather, the np.s_ object works without numpy

Reply:
I will try to parse all the inputs as np.s_ objects, it could work.

Reply:
Or can try to parse the np.s_ object to the gather. The other approach failed.

Reply:
Okay so I can just make the slices arrays and then gather works fine.